src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final int INITIAL_VALUE = 0;
+    private static final int OPERATION_PART = 0;
+    private static final int TRANSACTION_AMOUNT_PART = 1;
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String REPORT_SUPPLY = "supply";
+    private static final String REPORT_BUY = "buy";
+    private static final String REPORT_RESULT = "result";
+    private static final String SEPARATOR = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        int supply = INITIAL_VALUE;
+        int buy = INITIAL_VALUE;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String csvData;
+
+            while ((csvData = bufferedReader.readLine()) != null) {
+                String[] dataParts = csvData.split(",");
+
+                if (dataParts.length != 2) {
+                    throw new RuntimeException("Malformed CSV line: " + csvData);
+                }
+
+                String operation = dataParts[OPERATION_PART];
+                int transactionAmount;
+                try {
+                    transactionAmount = Integer.parseInt(dataParts[TRANSACTION_AMOUNT_PART]);
+                } catch (NumberFormatException e) {
+                    throw new RuntimeException("Invalid number in CSV line: " + csvData, e);
```

You parse the amount from the raw token directly and wrap NumberFormatException, but you don't trim the token and the thrown RuntimeException message lacks file context. Trim the amount token before parsing and wrap the exception with a meaningful message including `fromFileName` and the offending line, e.g. `throw new RuntimeException("Can't parse amount in file " + fromFileName + " line: " + csvData, e);`. The checklist requires meaningful exception handling (see "Don't ignore exceptions").